program/rust/src/rust_oracle.rs

```diff
@@ -803,3 +836,77 @@ pub fn del_product(
 
     Ok(())
 }
+
+
+pub fn upd_permissions(
```

Comment:
Could you add a comment explaining the semantics of `upd_permissions`, like there are on the other public functions (e.g. `del_product`)? In general docstrings on top-level public functions are very valuable.